asset/encoding.go

```diff
@@ -596,6 +604,10 @@ func LeafEncoder(w io.Writer, val any, buf *[8]byte) error {
 }
 
 func LeafDecoder(r io.Reader, val any, buf *[8]byte, l uint64) error {
+	if l > MaxAssetEncodeSize {
+		return tlv.ErrRecordTooLarge
+	}
+
```

Comment:
We basically have that with `DecodeP2P`. But we can't use that here because some records within the structs are allowed to be bigger, so we need to do it manually. But I agree, if we use TLV everywhere, then that would be a nice feature to have baked into the library.

Reply:
Didn't know about `DecodeP2P`, thanks.